### PR TITLE
[Product] Add review

### LINKED ISSUE #32: As a Customer, I want to submit a product review
* Submit a review for a product

```
POST /products/{slug}/reviews
```

Parameters:

* E-mail (optional)
* Name
* Review

### PR DESCRIPTION
**Pull request by lchrusciel**
Resolves #32

### CODE REVIEW

File: spec/Handler/AddReviewHandlerSpec.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        ProductReviewRepositoryInterface $productReviewRepository,
+        ChannelRepositoryInterface $channelRepository,
+        ProductRepositoryInterface $productRepository,
+        ProductReviewerProviderInterface $productReviewerProvider,
+        ReviewFactoryInterface $reviewFactory
+    ) {
+        $this->beConstructedWith($productReviewRepository, $channelRepository, $productRepository, $productReviewerProvider, $reviewFactory);
+    }
+
+    function it_is_initializable()
```
Comment (pamil) [inline spec/Handler/AddReviewHandlerSpec.php]: Tbh we don't need this one.

File: spec/Provider/CustomerProviderSpec.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+use Sylius\Component\Core\Repository\CustomerRepositoryInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\ShopApiPlugin\Provider\CustomerProvider;
+use Sylius\ShopApiPlugin\Provider\CustomerProviderInterface;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+
+final class CustomerProviderSpec extends ObjectBehavior
+{
+    function let(CustomerRepositoryInterface $customerRepository, FactoryInterface $customerFactory) {
```
Comment (pamil) [inline spec/Provider/CustomerProviderSpec.php]: CS

File: tests/Controller/ProductAddReviewApiTest.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        $this->loadFixturesFromFile('shop.yml');
+        $this->loadFixturesFromFile('customer.yml');
+
+        $data =
+<<<EOT
+        {
+            "channelCode": "WEB_GB",
+            "title": "Awesome product",
+            "rating": 5,
+            "comment": "If I were a mug, I would like to be like this one!",
```
Comment (pamil) [inline tests/Controller/ProductAddReviewApiTest.php]: Creepy.

Comment (lchrusciel) [inline tests/Controller/ProductAddReviewApiTest.php]: 🐋

**Pull request merged**